jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/config/JmxScraperConfigFactory.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jmxscraper.config;
+
+import static io.opentelemetry.contrib.jmxscraper.util.StringUtils.isBlank;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+@SuppressWarnings({"SystemOut", "SystemExitOutsideMain"})
+public class JmxScraperConfigFactory {
```

Comment:
[minor] this factory could have been moved to a static method factory in JmxScraperConfig to avoid having a separate class just for parsing it.

Reply:
Ok, then let's keep it as-is for now, we can always revisit this later.